src/wallet.cpp

```diff
@@ -1704,8 +1707,17 @@ bool CWallet::SelectStakeCoins(std::set<std::pair<const CWalletTx*, unsigned int
         if (out.tx->vout[out.i].nValue < nStakeAmount)
             continue;
 
+        //For Bulwark Re-staking since we know if a tx is previous stake it'll be considered re-stake resulting in various advantages
```

Comment:
this adds more incentive to stake grind, but in a different way,might wanna see how this plays out on a testnet